app/controllers/posts_controller.rb

```diff
@@ -1,20 +1,20 @@
 class PostsController < ApplicationController
   skip_before_action :verify_authenticity_token
   def index
-    @user = User.find(params[:user_id])
-    @posts = @user.posts
+    @user = User.includes(:posts).find(params[:user_id])
   end
 
   def show
-    @post = Post.find(params[:id])
+    @user = User.find(params[:user_id])
+    @post = @user.posts.includes(:comments, :likes).find(params[:id])
   end
 
   def new
-    @post = Post.new
+    # @post = Post.new
   end
```

Comment:
OPTIONAL
- [ ] Uncomment this line so that the new action can be successfully routed.

Reply:
Thank you for letting me know. I uncommented it.